src/libraries/System.Net.Sockets/tests/FunctionalTests/SocketDuplicationTests.cs

```diff
@@ -0,0 +1,391 @@
+        {
+            SocketInformation socketInformation = default;
+            Assert.Throws<PlatformNotSupportedException>(() => new Socket(socketInformation));
+        }
+
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [Fact]
+        public void DuplicateAndClose_Unix_ThrowsPlatformNotSupportedException()
+        {
+            using Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp);
+            int processId = CurrentProcessId;
+
+            Assert.Throws<PlatformNotSupportedException>(() => socket.DuplicateAndClose(processId));
+        }
+
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [Fact]
+        public async Task DuplicateAndClose_TcpClient()
+        {
+            using Socket listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            listener.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+            listener.Listen(1);
+
+            using Socket client0 = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+
+            using Socket client1 = new Socket(client0.DuplicateAndClose(CurrentProcessId));
+            Assert.False(client1.Connected);
+            client1.Connect(listener.LocalEndPoint);
+
+            using Socket client2 = new Socket(client1.DuplicateAndClose(CurrentProcessId));
+            Assert.True(client2.Connected);
+
+            using Socket handler = await listener.AcceptAsync();
+            await client2.SendAsync(TestBytes, SocketFlags.None);
+
+            int rcvCount = await handler.ReceiveAsync(_receiveBuffer, SocketFlags.None);
+
+            string receivedMessage = GetMessageString(_receiveBuffer, rcvCount);
+            Assert.Equal(TestMessage, receivedMessage);
+        }
+
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [Fact]
+        public async Task DuplicateAndClose_TcpListener()
+        {
+            using Socket listener0 = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            listener0.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+            listener0.Listen(1);
+
+            using Socket listener1 = new Socket(listener0.DuplicateAndClose(CurrentProcessId));
+
+            using Socket client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            await client.ConnectAsync(listener1.LocalEndPoint);
+
+            using Socket handler = await listener1.AcceptAsync();
+            await client.SendAsync(TestBytes, SocketFlags.None);
+
+            byte[] receivedBuffer = new byte[32];
+            int rcvCount = await handler.ReceiveAsync(new ArraySegment<byte>(receivedBuffer), SocketFlags.None);
+
+            string receivedMessage = GetMessageString(receivedBuffer, rcvCount);
+            Assert.Equal(TestMessage, receivedMessage);
+        }
+
+        [OuterLoop] // long-running
```

Comment:
What makes it long-running?

Reply:
For the same reason `CreateSocket.CtorAndAccept_SocketNotKeptAliveViaInheritance` is running for about 2 sec in debug on my PC. There are nested child processes + pipes. I'm also wondering why is it that much.

Reply:
We should figure out why it's taking 2 seconds.  It shouldn't be that long.

Reply:
Since this also concerns other tests, I'd rather take care of it after separately. I'm preparing a test fix PR anyways.